src/meteors/lime.py

```diff
@@ -362,351 +673,445 @@ def get_band_mask(
-
             dict_labels_to_segment_ids = {segment: idx + 1 for idx, segment in enumerate(band_names_hashed)}
             segments_list = band_names_hashed
         elif isinstance(band_names, dict):
-            logger.debug("band_names is a dictionary of segments")
-            dict_labels_to_segment_ids = band_names
-            segments_list = list(band_names.keys())
+            dict_labels_to_segment_ids = band_names.copy()
+            segments_list = tuple(band_names.keys())  # type: ignore
         else:
             raise ValueError("Incorrect band_names type. It should be a dict or a list")
 
         segments_list_after_mapping = [Lime._extract_bands_from_spyndex(segment) for segment in segments_list]
-        band_ranges_wavelengths: dict[str | tuple[str, ...], list[tuple[float, float]]] = {}
+        band_ranges_wavelengths: dict[tuple[str, ...] | str, list[tuple[float, float]]] = {}
         for segment in segments_list_after_mapping:
-            band_ranges_wavelengths[segment] = []
-            for band_name in segment:
-                min_wavelength = spyndex.bands[band_name].min_wavelength
-                max_wavelength = spyndex.bands[band_name].max_wavelength
-
-                band_ranges_wavelengths[segment].append((min_wavelength, max_wavelength))
+            band_ranges_wavelengths[segment] = [
+                (spyndex.bands[band_name].min_wavelength, spyndex.bands[band_name].max_wavelength)
+                for band_name in segment
+            ]
 
         return band_ranges_wavelengths, dict_labels_to_segment_ids
 
+    @staticmethod
+    def _convert_wavelengths_to_indices(
+        wavelengths: torch.Tensor, ranges: list[tuple[float, float]] | tuple[float, float]
+    ) -> list[tuple[int, int]]:
+        """Converts wavelength ranges to index ranges.
+
+        Args:
+            wavelengths (torch.Tensor): The tensor containing the wavelengths.
+            ranges (list[tuple[float, float]] | tuple[float, float]): The wavelength ranges.
+
+        Returns:
+            list[tuple[int, int]]: The index ranges corresponding to the wavelength ranges.
+        """
+        indices = []
+        if len(ranges) == 2 and all(isinstance(x, (int, float)) for x in ranges):
+            ranges = [tuple(ranges)]  # type: ignore
+
+        for start, end in ranges:  # type: ignore
+            start_idx = torch.searchsorted(wavelengths, start, side="right")
+            end_idx = torch.searchsorted(wavelengths, end, side="left")
+            indices.append((start_idx.item(), end_idx.item()))
+        return indices
+
     @staticmethod
     def _get_band_range_indices_from_band_ranges_wavelengths(
         image: Image,
-        band_ranges_wavelengths: dict[
-            str | tuple[str, ...],
-            list[tuple[float, float] | list[float]] | tuple[float, float] | list[float],
-        ],
+        band_ranges_wavelengths: dict[str | tuple[str, ...], list[tuple[float, float]] | tuple[float, float]],
```

Comment:
this types seems to be limited compared to my previous idea. I wanted to include the list of floats inside the outer list, instead of tuple of two floats

Reply:
Okey I understand so we can accept `list[tuple[float, float]]` or `tuple[float, float]` or `list[float]` (which is the list of bands) as a value to the dict, yes @Fersoil?